### PR TITLE
Add lsstfam option to closed loop.

### PR DESCRIPTION
**Pull request by jbkalmbach**
Debug rotation issues as well.

### CODE REVIEW

File: python/lsst/ts/imsim/opdMetrology.py [hunk 1]
```diff
@@ -145,20 +151,28 @@ def setWgtAndFieldXyOfGQ(self, instName):
...
+        fieldY = []
+        detMap = camera.getIdMap()
+        for detId in self.sensorIds:
+            detCenter = detMap[detId].getCenter(FIELD_ANGLE)
+            # Switch X,Y coordinates to convert from DVCS to CCS coords
+            fieldY.append(np.degrees(detCenter[0]))
+            fieldX.append(np.degrees(detCenter[1]))
+        self.fieldX = np.array(fieldX)
+        self.fieldY = np.array(fieldY)
+        # Convert from CCS to ZCS for current OFC
```
Comment (gmegh) [inline python/lsst/ts/imsim/opdMetrology.py:170]: Maybe also add the TO-DO comment here?

File: python/lsst/ts/imsim/opdMetrology.py [hunk 2]
```diff
@@ -89,6 +78,8 @@ def setDefaultLsstWfsGQ(self):
         self.wt = np.array([1.0, 1.0, 1.0, 1.0])
         wfsFieldX, wfsFieldY, sensorIds = self.getDefaultLsstWfsGQ()
         self.fieldX, self.fieldY = (wfsFieldX, wfsFieldY)
+        # Convert from CCS to ZCS for current OFC
+        self.fieldX = -1.0 * np.array(self.fieldX)
         self.sensorIds = sensorIds
 
     def getDefaultLsstWfsGQ(self):
```
Comment (gmegh) [inline python/lsst/ts/imsim/opdMetrology.py:85]: One general comment, I know it may take some time, but it might be nice to add typing to this package? If so, we could start in this PR with the functions here? Alternatively we can also do it in another PR?

File: python/lsst/ts/imsim/opdMetrology.py [hunk 3]
```diff
@@ -89,6 +78,8 @@ def setDefaultLsstWfsGQ(self):
         self.wt = np.array([1.0, 1.0, 1.0, 1.0])
         wfsFieldX, wfsFieldY, sensorIds = self.getDefaultLsstWfsGQ()
         self.fieldX, self.fieldY = (wfsFieldX, wfsFieldY)
+        # Convert from CCS to ZCS for current OFC
```
Comment (gmegh) [inline python/lsst/ts/imsim/opdMetrology.py:81]: Add TO-DO tag at the beginning of the comment?

File: python/lsst/ts/imsim/imsimCmpt.py [hunk 4]
```diff
@@ -277,14 +273,15 @@ def addConfigHeader(self, obsMetadata):
             Header information for ImSim config
         """
         headerText = "  header:\n"
-        headerText += f"    mjd: {obsMetadata.mjd}\n"
+        headerText += f"    mjd: *mjd\n"
         headerText += f"    observationStartMJD: {obsMetadata.mjd - (15/(60*60*24))}\n"
-        headerText += f"    seqnum: {obsMetadata.seqNum}\n"
-        headerText += f"    band: {obsMetadata.band}\n"
+        headerText += f"    seqnum: *seqnum\n"
```
Comment (gmegh) [inline python/lsst/ts/imsim/imsimCmpt.py:278]: Out of curiosity, is this using star unpacking of obsMetadata? I am guessing this can't be used for the other fields, right?

**Review by gmegh — CHANGES_REQUESTED** [review]
It is looking good. One general question is whether we want to remove comcam altogether since we might need it if it goes back on-sky. Maybe it can be added in another PR, however. Thanks!

**Comment by jbkalmbach** [comment]
Yeah, at the meeting on Friday I mentioned ComCam and Josh said the imSim images from ComCam can't be ingested by the butler right now but when they are we can add in the ComCam functionality in a separate issue/PR.

File: python/lsst/ts/imsim/opdMetrology.py (hunk 2 above)
Comment (jbkalmbach) [reply python/lsst/ts/imsim/opdMetrology.py:85]: Added it to the list of things in this issue: https://jira.lsstcorp.org/browse/DM-40355?filter=21514

Comment (jbkalmbach) [reply python/lsst/ts/imsim/opdMetrology.py:85]: If you have more formatting comments let's add them there.

File: python/lsst/ts/imsim/opdMetrology.py (hunk 3 above)
Comment (jbkalmbach) [reply python/lsst/ts/imsim/opdMetrology.py:81]: Done

File: tests/test_opdMetrology.py [hunk 5]
```diff
@@ -149,8 +66,24 @@ def testSetDefaultLsstWfsGQ(self):
 
         fieldX = self.metr.fieldX
         fieldY = self.metr.fieldY
-        self.assertCountEqual(fieldX, [1.176, -1.176, -1.176, 1.176])
-        self.assertCountEqual(fieldY, [1.176, -1.176, -1.176, 1.176])
+        self.assertCountEqual(
+            fieldX,
+            [
+                1.1902777777777778,
```
[the remark this answers is not in the document]
Comment (jbkalmbach) [reply tests/test_opdMetrology.py:72]: It was kind of hard to make a test different from the function without specifying values but I tried to at least calculate the mean values from obs_lsst a different way.

File: python/lsst/ts/imsim/imsimCmpt.py (hunk 4 above)
Comment (jbkalmbach) [reply python/lsst/ts/imsim/imsimCmpt.py:278]: No, when loading the yaml from a string it will fill in these values with values from `evalVariables`. We could just load them all from the `obsMetadata` object if that's easier to understand.

Comment (gmegh) [reply python/lsst/ts/imsim/imsimCmpt.py:278]: I think it’s fine

**Review by gmegh — APPROVED** [review]
Looks good! Thanks :)
